espnet2/bin/tts_inference.py

```diff
@@ -123,6 +131,56 @@ def __init__(
             self.spc2wav = None
             logging.info("Vocoder is not used because vocoder_conf is not sufficient")
 
+    def from_pretrained(filename_or_model_id: str, **kwargs) -> "Text2Speech":
+        """Instantiate a Text2Speech model from a local packed archive or a model id
+
+        Args:
+            filename_or_model_id (str): Path to a local packed archive, or model id from
+                the huggingface.co model hub
+                (e.g. ``"julien-c/kan-bayashi-jsut_tts_train_tacotron2_ja"``)
+
+        Returns:
+            instance of Text2Speech
+
+        """
+        if os.path.isfile(filename_or_model_id):
+            outpath = os.path.dirname(filename_or_model_id)
+            inputs = unpack(input_archive=filename_or_model_id, outpath=outpath)
+        else:
+            # If not found locally, let's try to find it on Hugging Face model hub
+            # e.g. julien-c/model is a valid model id
+            # and  julien-c/model@main supports specifying a commit/branch/tag.
+            if "@" in filename_or_model_id:
+                model_id = filename_or_model_id.split("@")[0]
+                revision = filename_or_model_id.split("@")[1]
+            else:
+                model_id = filename_or_model_id
+                revision = None
+            meta_yaml_url = hf_hub_url(
+                model_id, filename=META_YAML_FILENAME, revision=revision
+            )
+            meta_yaml_path = cached_download(
+                meta_yaml_url, library_name="espnet", library_version=__version__
+            )
+            with open(meta_yaml_path, "r", encoding="utf-8") as f:
+                d = yaml.safe_load(f)
+            assert isinstance(d, dict), type(d)
+            yaml_files = d["yaml_files"]
+            files = d["files"]
+            assert isinstance(yaml_files, dict), type(yaml_files)
+            assert isinstance(files, dict), type(files)
+            inputs = {}
+            for key, value in list(yaml_files.items()) + list(files.items()):
+                file_url = hf_hub_url(model_id, filename=value, revision=revision)
+                inputs[key] = cached_download(
+                    file_url, library_name="espnet", library_version=__version__
+                )
+                if key in yaml_files.keys():
+                    # Rewrite paths inside yaml
+                    hf_rewrite_yaml(inputs[key], model_id=model_id, revision=revision)
```

Comment:
I think this line is wrong, there is no revision param at the moment 

Trying this fails. 

```
!pip install -e git+https://github.com/Fhrozen/espnet.git@pr-huggingface#egg=espnet
from espnet2.bin.tts_inference import Text2Speech
    
model = Text2Speech.from_pretrained("espnet/kan_bayashi_jsut_tts_train_conformer_fastspeech2_raw_phn_jaconv_pyopenjtalk_train.loss.ave")

speech, *_ = model("text to generate speech from")
```

Reply:
yes, I am modifying the code given some codes. I have this fixed on some local commits. but I will push it when https://github.com/espnet/espnet_model_zoo/pull/39 is active

Reply:
Very cool! I just tested with latest changes and runs well! (probably I didn't pick a right model, but it's good to see this works e2e)

```

!pip install -e git+https://github.com/Fhrozen/espnet_model_zoo.git@pr-huggingface#egg=espnet_zoo
!pip install -e git+https://github.com/Fhrozen/espnet.git@pr-huggingface#egg=espnet
!pip install pyopenjtalk

from espnet2.bin.tts_inference import Text2Speech
    
model = Text2Speech.from_pretrained("espnet/kan_bayashi_jsut_tts_train_conformer_fastspeech2_raw_phn_jaconv_pyopenjtalk_train.loss.ave")
speech, *_ = model("text to generate speech from")

from IPython.display import display, Audio
display(Audio(speech.cpu().numpy(), rate=model.fs))
```